llvm/lib/Analysis/DXILResource.cpp

```diff
@@ -782,9 +781,8 @@ DXILBindingMap::findByUse(const Value *Key) const {
   }
 
   const CallInst *CI = dyn_cast<CallInst>(Key);
```

Comment:
How does it impact the recursion if we check for CallInst before PHINode? I kind of like my base cases to be first, but I know thats not always possible.

Reply:
Happy to test it but it looks like `PHINode` isn't a subclass of `CallInst` right? https://llvm.org/doxygen/classllvm_1_1PHINode.html

Which means the base case would earily exit before the PHINode code had a chance to run.  Could maybe tuck it inside the `if (!CI)`  but that feels a bit messsy to me

Reply:
yeah the trivial reorder causes the PHINodes case to fail